core/src/toga/app.py

```diff
@@ -525,6 +521,12 @@ def widgets(self) -> WidgetRegistry:
         """
         return self._widgets
 
+    @property
+    def windows(self) -> Iterable[Window]:
+        """The windows managed by the app. Windows are added to the app when they are
+        created, and removed when they are closed."""
```

Comment:
Now that there's no need for the user to manually add or remove windows from the app, the public type of this property can simply be `Iterable`, and `WindowSet` can be an implementation detail.